File: src/Messages/RegisteringMessages.cs
using System;
using Akka.Actor;

namespace Messages.CSharp
{
    public abstract partial class Message
    {

        public class RegisterPlayer
        {
            public string Name { get; private set; }

            public RegisterPlayer(string name)
            {
                Name = name;
            }
        }

        public class RegisterPlayerResponse
        {
            public Guid Token { get; private set; }

            public bool IsValid { get; private set; }

            public string Errors { get; private set; }

            public RegisterPlayerResponse(Guid token, bool isValid, string errors = null)
            {
                Token = token;
                IsValid = isValid;
                Errors = errors;
            }
        }

        public class UnregisterPlayer
        {
            public Guid Token { get; private set; }

            public UnregisterPlayer(Guid token)
            {
                Token = token;
            }
        }

        public class CreatePlayer
        {
            public Props Props { get; private set; }

            public CreatePlayer(Props props)
            {
                Props = props;
            }
        }
    }
}

File: src/Messages.FSharp/RegisteringMessages.fs
namespace Messages.FSharp
    open System

    type RegisterPlayer(name:string) =
        member this.Name = name

    type RegisterPlayerResponse(token:Guid, isValid:bool, errors:string) =
        member this.Token = token
        member this.IsValid = isValid
        member this.Errors = errors

    type UnregisterPlayer(token, gameToken) = 
        inherit GameMessage(token, gameToken)

    type CreatePlayer(name:string) =
        member this.Name = name
